multi-stage-query/src/main/java/org/apache/druid/msq/dart/worker/DartWorkerContext.java

```diff
@@ -128,6 +129,11 @@ public class DartWorkerContext implements WorkerContext
     this.tempDir = tempDir;
     this.queryContext = Preconditions.checkNotNull(queryContext, "queryContext");
     this.emitter = emitter;
+    
+    // Compute thread count once in constructor
+    final int baseThreadCount = processingConfig.getNumThreads();
+    final Integer maxThreads = MultiStageQueryContext.getMaxThreads(queryContext);
```

Comment:
We should probably have a -1 check since this property is user controlled.

Reply:
I was just thinking about how to handle that. IMO, it's reasonable to treat an invalid number like 0 or -1 the same as `null`. I just changed it.